contracts/external/proxies/LiquidatorProxyV1ForSoloMargin.sol

```diff
@@ -406,8 +413,65 @@ contract LiquidatorProxyV1ForSoloMargin is
         return supplyValue >= borrowValue.add(requiredMargin);
     }
 
-    // ============ Parsing Functions ============
+    // ============ Getter Functions ============
+
+    /**
+     * Gets the current total supplyValue and borrowValue for some account. Takes into account what
+     * the current index will be once updated.
+     */
+    function getCurrentAccountValues(
+        Constants memory constants,
+        Account.Info memory account
+    )
+        private
+        view
+        returns (
+            Monetary.Value memory,
+            Monetary.Value memory
+        )
+    {
+        Monetary.Value memory supplyValue;
+        Monetary.Value memory borrowValue;
+
+        for (uint256 m = 0; m < constants.markets.length; m++) {
+            Types.Par memory par = SOLO_MARGIN.getAccountPar(account, m);
```

Comment:
Wouldn't `SoloMargin#getAccountWei` do this for you...? It looks like that calls `fetchNewIndex`

Reply:
Yes, but that would require another call to the interestrate setter, etc. We already have the updated index so we just use that to save some gas.